src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -345,8 +349,11 @@ def _get_uniquely_named_nec_pnads(
         uniquely_named_nec_pnads: List[PartialNSRTAndDatastore] = []
         for pnad_list in sorted(param_opt_to_nec_pnads.values(), key=str):
             for i, pnad in enumerate(pnad_list):
-                pnad.op = pnad.op.copy_with(name=(pnad.op.name + str(i)))
-                uniquely_named_nec_pnads.append(pnad)
+                new_op = pnad.op.copy_with(name=(pnad.op.name + str(i)))
```

Comment:
without this change, checking harmlessness modifies the PNAD name in place, leading to the name getting extended after each iteration